src/js/popup.js

```diff
@@ -339,6 +342,98 @@ $(document).ready(function () {
         });
     });
 
+    $('#customDomains').click(function () {
+        dmUIClick('customDomains');
+        if (!$(this).hasClass('disabled')) {
+            $('.customDomainList').toggle();
+            $('main').toggle();
+        } else {
+            Materialize.toast(disabledOptionToast, 2000, 'toastNotification');
+        }
+    });
+
+    $('#customDomainsBackButton').click(function () {
+        dmUIClick('customDomainsBackButton');
+        $('.customDomainList').toggle();
+        $('main').toggle();
+    });
+
+    $('#clearCustomDomains').click(function () {
+        dmUIClick('clearCustomDomains');
+        // remove all added domains:
+        chrome.runtime.sendMessage({
+            JDTIfunction: 'removeDomain',
+            domainName: '',
+            removeAll: true
+        }, function (response) {
+            if (response.status === 'success') {
+                loadTemplateEditor();
+                Materialize.toast('Domains successfully removed', 2000, 'toastNotification');
+            } else {
+                if (response.message) {
+                    dmError('clearCustomDomains', response.message);
+                    Materialize.toast(response.message, 2000, 'toastNotification');
+                } else {
+                    dmError('clearCustomDomains', 'generic');
+                    Materialize.toast('Something went wrong. Please try again.', 2000, 'toastNotification');
+                }
+            }
+        });
+    });
+
+    $('#customDomainInput').keyup(function (event) {
+        if (event.keyCode === 13) {
+            $('#customDomainInputButton').click();
+        }
+    });
+
+    $('#customDomainInputButton').click(function () {
+        dmUIClick('customDomainInputButton');
+        var domainName = $('#customDomainInput').val();
+
+        chrome.runtime.sendMessage({
+            JDTIfunction: 'addDomain',
+            domainName: domainName
+        }, function (response) {
+            if (response.status === 'success') {
+                $('#customDomainInput').val('');
+                loadTemplateEditor();
+                Materialize.toast('Domain successfully added', 2000, 'toastNotification');
+            } else {
+                if (response.message) {
+                    dmError('customDomainInputButton', response.message);
+                    Materialize.toast(response.message, 2000, 'toastNotification');
+                } else {
+                    dmError('customDomainInputButton', 'generic');
+                    Materialize.toast('Something went wrong. Please try again.', 2000, 'toastNotification');
+                }
+            }
+        });
+    });
+
+    $(document).on('click', '#customDomainRemoveButton', function () {
```

Comment:
`$('#customDomainRemoveButton').click(function () {`
To be consistent with the other one.

Reply:
It is consistent with the 'removeSingleTemplate'. I've been having some difficulty with this, most likely because the list is built in the sandbox (like 'removeSingleTemplate'). Right now I'm still using `.on('click'`, but I did update it so that it uses the class instead of the id (so we don't have identical IDs for each domain div).